contracts/apm/Repo.sol

```diff
@@ -0,0 +1,91 @@
+pragma solidity ^0.4.15;
+
+import "../zeppelin/lifecycle/Ownable.sol";
+import "../apps/AppStorage.sol";
+
+
+contract Repo is AppStorage, Ownable {
+    struct Version {
+        uint16[3] semanticVersion;
+        address contractAddress;
+        bytes contentURI;
+    }
+
+    Version[] versions;
+    mapping (bytes32 => uint256) versionIdForSemantic;
+    mapping (address => uint256) latestVersionIdForContract;
+
+    event NewVersion(uint256 versionId, uint16[3] semanticVersion);
+
+    /**
+    * @notice Create new version for repo
+    * @param _newSemanticVersion Semantic version for new repo version
+    * @param _contractAddress address for smart contract logic for version (if set to 0, it uses last versions' contractAddress)
+    * @param _contentURI External URI for fetching new version's content
+    */
+    function newVersion(uint16[3] _newSemanticVersion, address _contractAddress, bytes _contentURI) onlyOwner public {
```

Comment:
I wonder how this will work with permission escalation? Is there a particular reason that it is not an app?

There might be use cases where you want a group of people to be able to create versions/mange repositories.

Reply:
Problem with making it a full app is that the DAO kernel is the APM DAO's, therefore we would need to grant users permission over their Repo on that ACL.

Now that I'm really thinking about it, this wouldn't be that crazy (only that these DAO ACL will be yuge), and when the APM creates a repo it can set the permissions to create versions either to itself or the owner of the repo.